### PR TITLE
Don't create compendia for organisms that don't have a QN target.

### PR DESCRIPTION
**Pull request by kurtwheeler**
## Issue Number

https://github.com/AlexsLemonade/refinebio/pull/449

## Purpose/Implementation Notes

This was always supposed to be a thing we checked. Part of creating compendia is quantile normalizing them, so if we don't have a QN target we should:
  A. Not try to create a compendia.
  B. Fail to create one if we did try.

## Methods

@jaclyn-taroni will be happy to hear that we're no longer going to create compendia for organisms that don't have a QN target.

## Types of changes

- Bugfix (non-breaking change which fixes an issue)

## Functional tests

WIP

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by jaclyn-taroni** [comment]
Woo!

File: foreman/data_refinery_foreman/foreman/management/commands/create_compendia.py [hunk 1]
```diff
@@ -90,9 +90,12 @@ def handle(self, *args, **options):
         logger.debug(all_organisms)
 
         for organism in all_organisms:
-            logger.debug(organism)
-            job = create_job_for_organism(organism, quant_sf_only, svd_algorithm)
-            logger.info("Sending CREATE_COMPENDIA for Organism", job_id=str(job.pk), organism=str(organism))
-            send_job(ProcessorPipeline.CREATE_COMPENDIA, job)
+            if get_most_recent_qn_target_for_organism(organism):
```
Comment (arielsvn) [inline foreman/data_refinery_foreman/foreman/management/commands/create_compendia.py]: We have `organism.qn_target`

File: common/data_refinery_common/utils.py [hunk 2]
```diff
@@ -453,8 +454,25 @@ def queryset_page_iterator(queryset, page_size = 2000):
...
 
+
 def queryset_iterator(queryset, page_size = 2000):
     """ use the performant paginator to iterate over a queryset """
     for page in queryset_page_iterator(queryset, page_size):
         for item in page:
             yield item
+
+
+def get_most_recent_qn_target_for_organism(organism):
```
[the remark this answers is not in the document]
Comment (arielsvn) [reply common/data_refinery_common/utils.py]: Just noticed, can we use `organism.qn_target` instead of this?

**Comment by kurtwheeler** [comment]
Ok I think my latest commit addressed your comment. Here's hoping the tests pass :crossed_fingers:

File: common/data_refinery_common/models/organism.py [hunk 3]
```diff
@@ -187,5 +191,20 @@ def get_objects_with_qn_targets(cls):
         """ Return a list of Organisms who already have valid QN targets associated with them. """
         return Organism.objects.all().filter(qn_target__isnull=False)
 
+    def get_most_recent_qn_target(self):
```
Comment (arielsvn) [inline common/data_refinery_common/models/organism.py]: Do we need this method? I think we save the most recent QN Target in `organism.qn_target`

Comment (kurtwheeler) [reply common/data_refinery_common/models/organism.py]: Good point.... that solves the issue I was having with import rings.

Good catch, wish we had realized that earlier lol

**Comment by kurtwheeler** [comment]
I've made the requested changes on this PR and re-requested a review on it, why is its status still "Changes Requested"?

**Pull request merged**